### DISCUSSION TITLE
Advanced searchbar problems...

### DISCUSSION THREAD

**Discussion by RolandHesse** [neuefische/questions#785] [Web] [answered]
Hey there!

While we have now implemented a basic searchbar functionality, we still struggle with some issues. The latest of our attempts to solve them can be found here: https://github.com/PeetThesing/capstone-project_snippet-manager/pull/53

The problems we are still facing and were unable to solve till date are following:

1. Dropdown doesn't close. - Ideally, a dropdown menu showing the last searches is opened when clicking into the StyledSearchBarInput and closed when clicking outside the StyledLastSearchContainer. The former works, the latter doesn't - we are clueless as to why not. What we were able to implement is that the dropdown closes when clicking outside the StyledSearchBarInput - however, this is clearly not what we want, since, in this case, the user would not be able to select one of the last search terms via mouse click....

2. Accessibility. - Selecting one of the last search terms in the dropdown via the tab-key doesn't work at all. We do not know how to change this.

3. Filtering last searches when text is entered into the StyledSearchBarInput. - What we wanted to achieve is that, as soon as the user starts typing in the searchbar, the list of search terms is filtered so that only search terms that match with what has been typed into the searchbar are shown. We tried to use a derived state to implement this - alas, without success...

Any feedback is highly appreciated! 🙏

**Comment by doemser** [comment, marked as answer]
Several things I noticed on the spot

- You seem to control the search-field in order to offer fuzzy-search type of user-experience.. but your search history only appears on entering the searchterm - these two things seem like a contradiction and is the biggest issue for me, this leads to me not understanding how you want this component to behave..
- You have no clickable element in the history list
- The latest search overwrites the last one, so you only have one history entry

This is a rather hard to explain ticket, so I made you a prototype that essentially does what you want to achieve:

https://codesandbox.io/s/mystifying-snowflake-y3t434?file=/src/App.js

But keep in mind that I find it SUPER unintuitive.

**Reply by doemser** [reply]
Oh wait, I just found a bug in there

**Reply by doemser** [reply]
Fixed it. There you go.

**Comment by NadjaKanunnikow** [comment]
Thank you for your input, we tried something else and now the searchBar is finally in the code review. We will tackle the issue with the accessibility part in another user story.
